ui/web.go

```diff
@@ -72,8 +73,16 @@ func Register(r *route.Router, reloadCh chan<- struct{}, logger log.Logger) {
 	))
 
 	r.Post("/-/reload", func(w http.ResponseWriter, req *http.Request) {
-		w.Write([]byte("Reloading configuration file..."))
-		reloadCh <- struct{}{}
+		errc := make(chan error)
+		defer close(errc)
+
+		reloadCh <- errc
+		if err := <-errc; err != nil {
+			http.Error(w, fmt.Sprintf("config file reload failed: %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		w.Write([]byte("config file reloaded"))
```

Comment:
An empty 200 response is probably just as good. What does the Prometheus server do? I would out of consistency do the same.

Reply:
Same link from above: https://github.com/prometheus/prometheus/blob/master/web/web.go#L638-L644

Looks like we're only writing something on failure